sycl/include/syclcompat/math.hpp

```diff
@@ -151,6 +151,19 @@ template <typename T> sycl::vec<int32_t, 2> extractAndExtend2(T a) {
   return ret;
 }
 
+template <typename T> sycl::vec<int16_t, 4> extractAndExtend4(T a) {
+  sycl::vec<int16_t, 4> ret;
+  sycl::vec<T, 1> va{a};
+  using Tint =
```

Comment:
```suggestion
  using ByteT =
```
I think `ByteT` might be better, WDYT?

Reply:
Yeah was thinking about naming it after "type of int" as singed/unsigned. But sure I'll go with something like "IntT"